tests/api/core/content-manager/api/number-of-draft-relations.test.api.js

```diff
@@ -397,13 +430,65 @@ describe('CM API - Number of draft relations', () => {
       },
     });
 
+    nonDefaultLocaleProduct = localisedProduct;
+
     // Ensure we can count the number of draft relations when the entry is in a non default locale
     const { body } = await rq({
       method: 'GET',
       url: `/content-manager/collection-types/api::product.product/${localisedProduct.documentId}/actions/countDraftRelations`,
-      qs: { locale },
+      qs: { locale: nonDefaultLocale },
     });
 
     expect(body.data).toBe(8);
   });
+
+  test('Correctly count the number of draft relations across multiple locales and document IDs', async () => {
+    // IDs of the categories to use from the default locale
+    const publishedId = categories.published[0].id;
+    const draft1Id = categories.draft[0].id;
+    const draft2Id = categories.draft[1].id;
+
+    const documentId = nonDefaultLocaleProduct.documentId;
+
+    // Create a product in the default locale of the same product document as
+    // the non default locale product
+    await rq({
+      method: 'PUT',
+      url: `/content-manager/collection-types/api::product.product/${documentId}`,
+      // With 4 draft relations
+      body: {
+        name: 'Pizza',
+        categories: [publishedId],
+        compo: {
+          onecategory: draft1Id,
+          categories: [draft2Id],
+        },
+        comporep: [
+          {
+            categories: [draft2Id],
+          },
+        ],
+        dz: [
+          {
+            __component: 'default.compo',
+            categories: [draft1Id],
+          },
+        ],
+      },
+    });
+
+    const documentIds = [documentId, defaultLocaleDocument.documentId];
+    const { body } = await rq({
+      method: 'GET',
+      url: `/content-manager/collection-types/api::product.product/actions/countManyEntriesDraftRelations`,
+      qs: {
+        documentIds,
+        locale: ['en', 'fr'],
+      },
+    });
+
+    // 4 (newly created default locale) + 8 (non default locale of the same
+    // document) + 12 (previously created default locale) = 24
+    expect(body.data).toBe(24);
```

Comment:
Since this depends on previous tests running, you can't use `.only` on this test and if anyone adds a test they will have to update this. Could it do a db reset, create some data, then count that?